ansible_base/authentication/authenticator_plugins/saml.py

```diff
@@ -289,8 +289,16 @@ def extra_data(self, user, backend, response, *args, **kwargs):
             if perm in attrs:
                 kwargs["social"].extra_data[perm] = attrs[perm]
 
-        # Move group spec up a level if present
-        if "Group" in attrs:
+        # Get configured group attribute, if present
+        configuration = self.database_instance.configuration if hasattr(self.database_instance, 'configuration') else None
+        idp_groups_attribute_name = self.configuration_class.settings_to_enabled_idps_fields['IDP_GROUPS']
+        configured_groups_attribute = configuration['ENABLED_IDPS'][idp_string].get(idp_groups_attribute_name) if configuration is not None else None
+        if configured_groups_attribute is not None and configured_groups_attribute in attrs:
+            logger.debug(f"Setting Group from attribute: {configured_groups_attribute}")
+            response["Group"] = attrs[configured_groups_attribute]
+        # Else try getting the "Group" attribute, if present
+        elif "Group" in attrs:
+            logger.debug("Setting Group from attribute: Group")
             response["Group"] = attrs["Group"]
```

Comment:
```suggestion
        self.groups_claim = configured_groups_attribute if configured_groups_attribute else self.groups_claim

        if self.groups_claim in attrs:
            response[self.groups_claim] = attrs[self.groups_claim]
```

Reply:
I think we want Group hard coded (@tznamena correct me if I'm wrong)
But we could make this even more compact:
```suggestion
        response["Group"] = attrs.get(configured_groups_attribute, attrs.get(self.groups_claim, []))
```

This would also give the benefit that if a user specified an attribute but it comes back as empty we would still fall back to the normal Groups.

Reply:
We already initialize super().groups_claim to "Group", so if this doesn't modify it, it will be Group.  Also, the other code already looks for the group list in response[groups_claim] so it seems odd to me to force it into response["Group"] regardless of the configuration..

I definitely like the compactness though. :) 

class SocialAuthMixin:
    configuration_encrypted_fields = []
    logger = None
    groups_claim = "Group"